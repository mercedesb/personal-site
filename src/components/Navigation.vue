<template>
  <nav class="Navigation">
    <ul class='Navigation-navLinks' v-if="navLinks.length">
      <li class="Navigation-navLink">
        <smart-link to="/">
          <div class='Navigation-title'><span>Home</span></div>
          <img class='Navigation-icon' src='../assets/HomeIcon.svg' />
        </smart-link>
      </li>
      <li class="Navigation-navLink" v-for="navLink in navLinks" :key="navLink.title">
        <smart-link
          :to="navLink.parsedLink"
          :isExternal="!!navLink.externalLink"
          >
          <div class='Navigation-title'><span>{{navLink.title}}</span></div>
          <img class='Navigation-icon' :src="navLink.iconUrl" />
        </smart-link>
      </li>
    </ul>
  </nav>
</template>

<script>
import images from '../mixins/images'

export default {
  mixins: [
    images
  ],
  computed: {
    navLinks () {
      return this.$store.state.navLinks.map((link) => {
        return {
          parsedLink: link.externalLink ? link.externalLink : `/${link.urlSegment}`,
          iconUrl: this.getImageUrl(link.icon),
          ...link
        }
      })
    }
  }
}
</script>

<style lang="scss">
@import '../assets/styles/variables.scss';

$transition-time: .3s;

$icon-initial-width: 50px;
$icon-hover-width: 75px;

.Navigation {
  display: flex;

  &-navLinks {
    margin-bottom: 0;
    display: flex;
    align-items: flex-end;
    justify-content: space-between;

    z-index: 1;
  }

  &-navLink {
    margin: 0 0 $small-spacing;

    a {
      display: flex;
      flex-direction: column;
      align-items: center;

      @include media($min-tablet) {
        margin: 0 $small-spacing 0;
        padding-bottom: $small-spacing / 2;

        &:hover {
          cursor: pointer;

          .Navigation {
            &-title {
              display: block;
            }

            &-icon {
              max-width: $icon-hover-width;
              -webkit-transition: max-width $transition-time;
              transition: max-width $transition-time;
            }
          }
        }

        &.router-link-exact-active {
          border-bottom: 1px solid $white;
        }
      }
    }
  }

  &-title {
    display: none;
    padding-bottom: $small-spacing;
    color: $white;
  }

  &-icon {
    margin: 0 $small-spacing;
    max-width: $icon-initial-width;
    -webkit-transition: max-width $transition-time;
    transition: max-width $transition-time;
  }
}
</style>
