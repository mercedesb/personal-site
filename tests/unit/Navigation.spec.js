import Navigation from "@/components/Navigation.vue";
import { TestUtility } from "./TestUtility";

describe("Navigation", () => {
  let component;

  const homePageNavLink = {
    icon: {
      fields: {
        svg: "<svg></svg>"
      }
    }
  };

  const landingPageNavLinks = Object.keys(TestUtility.landingPages).map(key => {
    return TestUtility.landingPages[key];
  });

  const initialProps = {
    navLinks: landingPageNavLinks,
    homeLink: homePageNavLink
  };

  const shallow = propsData =>
    TestUtility.shallow(Navigation, {
      propsData: {
        ...initialProps,
        ...propsData
      }
    });

  describe("Snapshots", () => {
    it("matches snapshot", () => {
      component = shallow();
      expect(component.element).toMatchSnapshot();
    });
  });
});
