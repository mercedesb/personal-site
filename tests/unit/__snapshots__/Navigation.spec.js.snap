// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Snapshots matches snapshot 1`] = `
<nav
  class="Navigation"
>
  <ul
    class="Navigation-navLinks"
  >
    <li
      class="Navigation-navLink"
    >
      <smart-link-stub
        to="/"
      >
        <div
          class="Navigation-title"
        >
          <span>
            Home
          </span>
        </div>
         
        <div
          class="Navigation-icon"
        />
      </smart-link-stub>
    </li>
     
    <li
      class="Navigation-navLink"
    >
      <smart-link-stub>
        
        About
        
        <div
          class="Navigation-icon"
        >
          {
  "fields": {
    "svg": "
          <svg />
          "
  }
}
        </div>
      </smart-link-stub>
    </li>
    <li
      class="Navigation-navLink"
    >
      <smart-link-stub>
        
        Blog
        
        <div
          class="Navigation-icon"
        >
          {
  "fields": {
    "svg": "
          <svg />
          "
  }
}
        </div>
      </smart-link-stub>
    </li>
    <li
      class="Navigation-navLink"
    >
      <smart-link-stub>
        
        Connect
        
        <div
          class="Navigation-icon"
        >
          {
  "fields": {
    "svg": "
          <svg />
          "
  }
}
        </div>
      </smart-link-stub>
    </li>
    <li
      class="Navigation-navLink"
    >
      <smart-link-stub
        isexternal="true"
      >
        
        Shop
        
        <div
          class="Navigation-icon"
        >
          {
  "fields": {
    "svg": "
          <svg />
          "
  }
}
        </div>
      </smart-link-stub>
    </li>
  </ul>
</nav>
`;
